drivers/soundwire/intel_init.c

```diff
@@ -263,6 +267,25 @@ static struct sdw_intel_ctx
 		md->driver->probe(md, link);
 
 		list_add_tail(&link->list, &ctx->link_list);
+		bus = &link->cdns->bus;
+		/* Calculate number of slaves */
+		list_for_each_entry(slave, &bus->slaves, node)
+			num_slaves++;
```

Comment:
btw,  I assume you have checked there is no list API to get the list size ? It seems a common use.

Reply:
> btw, I assume you have checked there is no list API to get the list size ? It seems a common use.

Yeah, I searched https://www.kernel.org/doc/html/latest/core-api/kernel-api.html#list-management-functions but didn't find an useful API.